angr/analyses/decompiler/utils.py

```diff
@@ -319,6 +319,13 @@ def first_nonlabel_statement(block: ailment.Block) -> Optional[ailment.Stmt.Stat
     return None
 
 
+def last_nonlabel_statement(block: ailment.Block) -> Optional[ailment.Stmt.Statement]:
+    for stmt in block.statements[::-1]:
```

Comment:
Don't make a copy of `block.statements` when reverting it. It's a list, so you can just access the last item or use a revert generator.